tests/Microsoft.Experimental.Collections.Tests/Microsoft/Collections/Extensions/OrderedDictionaryTests.cs

```diff
@@ -0,0 +1,966 @@
+        }
+
+        [Fact]
+        public void Constructor_Succeeds_WhenCapacityIsZero()
+        {
+            OrderedDictionary<string, int> dictionary = new OrderedDictionary<string, int>(0);
+            dictionary = new OrderedDictionary<string, int>(0, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void Constructor_Succeeds_WhenCapacityIsGreaterThanZero()
+        {
+            OrderedDictionary<string, int> dictionary = new OrderedDictionary<string, int>(10);
+            dictionary = new OrderedDictionary<string, int>(10, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void Constructor_ThrowsArgumentOutOfRangeException_WhenCapacityIsLessThanZero()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new OrderedDictionary<string, int>(-1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new OrderedDictionary<string, int>(-1, StringComparer.OrdinalIgnoreCase));
+        }
+
+        [Fact]
+        public void Constructor_Succeeds_WhenComparerIsNull()
+        {
+            OrderedDictionary<string, int> dictionary = new OrderedDictionary<string, int>((IEqualityComparer<string>)null);
+            dictionary = new OrderedDictionary<string, int>(10, null);
+            dictionary = new OrderedDictionary<string, int>(new[] { KeyValuePair.Create("1", 1) }, null);
+        }
+
+        [Fact]
+        public void Constructor_Succeeds_WhenComparerIsNotNull()
+        {
+            OrderedDictionary<string, int> dictionary = new OrderedDictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+            dictionary = new OrderedDictionary<string, int>(10, StringComparer.OrdinalIgnoreCase);
+            dictionary = new OrderedDictionary<string, int>(new[] { KeyValuePair.Create("1", 1) }, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void Constructor_Succeeds_WhenCollectionIsProvided()
+        {
+            var collection = Enumerable.Range(0, s_defaultCount).Select(i => KeyValuePair.Create(i.ToString(), i)).ToList();
+            OrderedDictionary<string, int> dictionary = new OrderedDictionary<string, int>(collection);
+            ValidateDefaultOrderedDictionaryIsContiguous(dictionary);
+            new OrderedDictionary<string, int>(collection, StringComparer.OrdinalIgnoreCase);
+            ValidateDefaultOrderedDictionaryIsContiguous(dictionary);
+        }
+
+        [Fact]
+        public void Constructor_ThrowsArgumentNullException_WhenCollectionIsNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => new OrderedDictionary<string, int>((IEnumerable<KeyValuePair<string, int>>)null));
+            Assert.Throws<ArgumentNullException>(() => new OrderedDictionary<string, int>(null, StringComparer.OrdinalIgnoreCase));
+        }
+
+        [Fact]
+        public void TestWithStringComparerIgnoreCase()
```

Comment:
Do you have tests for hash collision? OrdinalIgnoreCase could be used for a collision of 2 entries (although this test doesn't do that) but it would probably be worth having a custom comparer to test behavior when > 2 entries collide. Eg., repeat some of the general tests using a comparer that returns the same hashcode for everything.

Reply:
I did this for DictionarySlim, they can share: https://github.com/dotnet/corefxlab/commit/da5f15991f7a9778f65c948bd54a646adebc10a1